src/django_components/template_loader.py

```diff
@@ -56,18 +78,19 @@ def get_dirs(self) -> List[Path]:
             for app_config in apps.get_app_configs():
                 if not app_config.path:
                     continue
-                if not Path(component_dir).is_dir():
-                    continue
 
-                if path_is_relative_to(component_dir, app_config.path):
+                # NOTE: This applies only to absolute paths
+                if path_is_relative_to(component_dir, app_config.path) and Path(component_dir).is_dir():
                     curr_directories.add(Path(component_dir).resolve())
 
+            # NOTE: This applies only to relative paths
             if hasattr(settings, "BASE_DIR"):
-                path = (Path(settings.BASE_DIR) / component_dir).resolve()
+                path: Path = (Path(settings.BASE_DIR) / component_dir).resolve()
```

Comment:
This branch allows us to treat a relative path in `STATICFILES_DIRS` as relative to the project root. The question is whether we want to support this, since in docs we ask users to set the dirs as:

```py
STATICFILES_DIRS = [
    ...,
    os.path.join(BASE_DIR, "components"),
]
```

At first I was in favour as it could be nicer experience. However, looking at Django's docs on [STATICFILES_DIRS](https://docs.djangoproject.com/en/5.0/ref/settings/#staticfiles-dirs), they ask for full paths. So if we on the other hand ask users to put relative paths into `STATICFILES_DIRS`, it could lead to unexpected behaviour. Instead, we should expect `STATICFILES_DIRS` to have full paths only.

Which means we'd remove the feature of supplying relative paths to `STATICFILES_DIRS`. I think this is fine, as resolving the paths relative to `BASE_DIR` is simple:
```py
Path(settings.BASE_DIR) / component_dir).resolve()
```

Reply:
UPDATE: I've refactored the Loader to expect only full paths from `STATICFILES_DIRS`.